File: 1. 단계별로_풀어보기/10~19/14. 집합과 맵/2. 14425.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

N, M = map(int, input().split())
SArr = list(input().strip() for i in range(N))
testArr = list(input().strip() for i in range(M))

SSet = set(SArr)

result = 0
for word in testArr:
  result += 1 if word in SSet else 0

output(f"{result}")

File: 1. 단계별로_풀어보기/10~19/14. 집합과 맵/2. 14425.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const [N, M] = inputArr[0].split(' ').map(Number)

const SArr = inputArr.slice(1, N + 1)
const TestArr = inputArr.slice(N + 1)

// dictionary 에 넣기
const dict = {}
SArr.forEach(word => (dict[word] = true))

// 존재하는 갯수 세기
const result = TestArr.filter(word => dict[word]).length

// 출력
process.stdout.write(`${result}`)
